public/app/plugins/datasource/grafana-azure-monitor-datasource/components/deprecated/utils.tsx

```diff
@@ -0,0 +1,13 @@
+import { config } from '@grafana/runtime';
+import { gt, valid } from 'semver';
+
+import { AzureDataSourceSettings } from '../../types';
+
+export function isAppInsightsConfigured(options: AzureDataSourceSettings) {
+  return !!(options.jsonData.appInsightsAppId && options.secureJsonFields.appInsightsApiKey);
+}
+
+export function gtGrafana9() {
+  // AppInsights configuration will be removed with Grafana 9
+  return valid(config.buildInfo.version) && gt(config.buildInfo.version, '9.0.0-beta1');
```

Comment:
This is neat, just double checking, do we know if there's any risk people might have an old version of the code cached in the browser, while the backend is newer or not really?

Reply:
I guess that's a possible race condition (e.g. if you don't refresh grafana) but the backend is also checking this (developed in https://github.com/grafana/grafana/pull/45834/files#diff-9b26372138c349dc10be78a372ff5e87a9c9d21fbee6b17efe0c66e658d5fdfbR35) so they will still get the error just a bit uglier (something like "there is no data source for appInsights").

Reply:
Cool sounds good, so long as it doesn't totally break or whatever, it seems pretty reasonable to just require users refresh